Fatal error: exception Failure("variable foo can't be borrowed after it has already been mutably borrowed.")
